internal/integration/retryable_writes_prose_test.go

```diff
@@ -380,3 +381,207 @@ func TestRetryableWritesProse(t *testing.T) {
 		}
 	})
 }
+
+type crudOperation struct {
+	Name      string   `bson:"name"`
+	Arguments bson.Raw `bson:"arguments"`
+}
+
+type crudOutcome struct {
+	Error      bool               `bson:"error"` // only used by retryable writes tests
+	Result     interface{}        `bson:"result"`
+	Collection *outcomeCollection `bson:"collection"`
+}
+
+// run a CRUD operation and verify errors and outcomes.
+// the test description is needed to see determine if the test is an aggregate with $out
+func runCrudOperation(mt *mtest.T, testDescription string, operation crudOperation, outcome crudOutcome) {
```

Comment:
All additional code in `retryable_writes_prose_test.go` was copied from `crud_spec_test.go`, which was removed because it's redundant with the new CRUD unified spec tests.